src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
```

Comment:
```suggestion
public class Lottery {

    private final ColorSupplier colorSupplier;
    private final Random random;
    private static final int MAX_BALL_NUMBER = 100;

    public Lottery(ColorSupplier colorSupplier) {
        this.colorSupplier = colorSupplier;
        this.random = new Random();
    }

```
Lottery it' not a subclass of ColorSupplier and doesn't extends it. You should use association with ColorSupplier. Create field of type Random for using it in getRandomBall() method. Define variable for storing value of ball's max number.